connector/connect/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectStreamHandler.scala

```diff
@@ -49,21 +51,33 @@ class SparkConnectStreamHandler(responseObserver: StreamObserver[Response]) exte
     }
   }
 
-  def handlePlan(session: SparkSession, request: proto.Request): Unit = {
+  def handlePlan(session: SparkSession, request: Request): Unit = {
     // Extract the plan from the request and convert it to a logical plan
     val planner = new SparkConnectPlanner(request.getPlan.getRoot, session)
-    val rows =
-      Dataset.ofRows(session, planner.transform())
-    processRows(request.getClientId, rows)
+    val dataframe = Dataset.ofRows(session, planner.transform())
+    request.getPreferredResultType match {
+      case Request.ResultType.ArrowBatch =>
+        // check whether all data types are supported
```

Comment:
What is not supported?

Reply:
for example: CharType, VarcharType, UserDefinedType(like VectorUDT)

supported list: 

https://github.com/apache/spark/blob/1a90512f605c490255f7b38215c207e64621475b/sql/catalyst/src/main/scala/org/apache/spark/sql/util/ArrowUtils.scala#L38-L60

https://github.com/apache/spark/blob/7b8016a578f511d1c17b16393c487429ce08f132/python/pyspark/sql/pandas/types.py#L54-L120